frontend/src/Components/UserWebSocket.jsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+const UserWebSocket = ({ userId }) => {
+  const [ws, setWs] = useState(null);
+
+  useEffect(() => {
+    const connectWebSocket = () => {
+      const socket = new WebSocket(`ws://localhost:5174/ws/${userId}`);
```

Comment:
Have we tested the multi socket case? Not blocking but let's double check later (also new to using vanilla WS so I'll look into it as well)